crates/telemetry/src/encoder.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
```

Comment:
A bit of a bummer we can't just import to use this, and instead we'll need to maintain it.

Reply:
That's actually from the Prometheus text encoder, which I used as a base for the JSON one that I wrote myself. But I had to keep the license since the JSON encoder is therefore technically a derived work.